blog/_posts/2022-01-01-apis-under-control.md

```diff
@@ -0,0 +1,110 @@
+2. Operate a _[control loop][control-loop]_ to monitor all inputs to the system and converge the system state towards the setpoint (the Kubernetes [control plane][control-plane]).
+3. Allow any interested consumer to monitor inputs and make changes to the system and its setpoint ([watches][] and [operators][Operators]).
+
+In this way, Kubernetes does for computing workloads what cruise control (speed as a setpoint) or autopilot (destination as a setpoint) do for your automobile, except in an open and extensible fashion. (This reminds me of a [vulnerability in Fiat/Chrysler/Jeep automobiles][jeep] years ago that allowed remote attackers complete control of the vehicle. [Secure your clusters][secure-clusters], everyone!)
+
+## From prod to dev
+
+When it comes to the software tools we use to develop our own systems, however, they are usually not so dynamic. They typically operate on some fixed configuration inputs, require us to run them manually after we make changes, and produce relatively static output. They may make assumptions about prerequisites that need to be installed. Errors can be cryptic. Some tools and frameworks have gotten a little more dynamic with auto-reloading development web servers, but that's about it. And then, gaining insight into how the tool operates or what produced the result is a difficult, frustrating task. Finally, getting different tools to work together is a constant tangle of shell scripts and baling wire.
+
+With Tilt, we are working to change the state of development environments from the fragile, static, duct-taped approach to a dynamic open, extensible environment that responds to your changes and immediately takes action, keeping you in a state of flow.
+
+## Dev environments as APIs
+
+It's [no][t8] [secret][t3] [that][t4] [we're][t5] [big][t6] [fans][t7] of [Kubernetes][t1], the [control loop][nicks-kubecon], and the [API][t2]. We like it so much, we spent most of 2021 migrating and adapting Tilt to use the Kubernetes API machinery. It's not an analog, it literally uses the same building blocks as Kubernetes.
+
+To see for yourself, start Tilt in one shell:
+```
+tilt up
+```
+In another shell, tell `kubectl` about Tilt's API server:
+```
+export KUBECONFIG=~/.tilt-dev/config
+kubectx tilt-default
+```
+Now you can browse Tilt using kubectl:
+```
+kubectl api-resources
+```
+This outputs:
+```
+NAME                    SHORTNAMES     APIVERSION          NAMESPACED   KIND
+clusters                               tilt.dev/v1alpha1   false        Cluster
+cmds                                   tilt.dev/v1alpha1   false        Cmd
+# ...all the resource kinds Tilt knows...
+```
+Now ask to see the filewatch resources:
+```
+kubectl get filewatches
+```
+This outputs:
+```
+NAME                 CREATED AT
+configs:(Tiltfile)      2021-12-15T21:10:03Z
+image:tilt-avatar-api   2021-12-15T21:10:03Z
+image:tilt-avatar-web   2021-12-15T21:10:03Z
+```
+
```

Comment:
I feel like this needs more in-depth explanation to tie into the point we are making before about the awesomeness of APIs